zzzz_modular_occulus/code/modules/projectiles/guns/mods/mods.dm

```diff
@@ -16,4 +16,20 @@
 
 /datum/design/autolathe/diffuser
 	name = "Mekhane \"DIFFUSER\" micro lens"
-	build_path = /obj/item/gun_upgrade/barrel/diffuser
\ No newline at end of file
+	build_path = /obj/item/gun_upgrade/barrel/diffuser
+
+/obj/item/gun_upgrade/mechanism/funswitch
+	name = "Make Shift \"Fun Switch\" full auto disconnector"
+	desc = "The Solar ATF considers this \"fun switch\" to be the galaxy's smallest machine gun, so don't order one online unless you're lonely and really want to hang out with an SATF squad."
+	icon_state = "Funswitch"
+	rarity_value = 30
+
+/obj/item/gun_upgrade/mechanism/funswitch/New()
+	..()
+	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
+	I.weapon_upgrades = list(
+		GUN_UPGRADE_FULLAUTO = TRUE,	//Makes the Paco able to fire full auto when installed
+		GUN_UPGRADE_RECOIL = 2		//But also makes it horrific to control
```

Comment:
Is this enough? Did you load up a server and test the cone on this? Did you test it with other mods like ergo grips?

Reply:
Values updated as requested in coding discussion. Newest test shows ~90 degree cone of fire by the end of a normal magazine burst.